### ISSUE TITLE
iPXE bootstrap - 404

### ISSUE THREAD

**Issue by vancluever** [rancher/os#2320]
Pretty sure this is recent:

```
$ curl -IL http://releases.rancher.com/os/latest
HTTP/1.1 301 Moved Permanently
Date: Thu, 05 Apr 2018 21:25:07 GMT
Connection: keep-alive
Cache-Control: max-age=3600
Expires: Thu, 05 Apr 2018 22:25:07 GMT
Location: https://releases.rancher.com/os/latest
Server: cloudflare
CF-RAY: ****

HTTP/2 404 
date: Thu, 05 Apr 2018 21:25:07 GMT
content-type: application/xml; charset=UTF-8
content-length: 192
set-cookie: ****
x-guploader-uploadid: ****
access-control-allow-origin: *
access-control-expose-headers: Content-Type
expires: Thu, 05 Apr 2018 21:25:07 GMT
cache-control: private, max-age=0
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: ****
```

This is producing error https://ipxe.org/3c00e103 in iPXE.

**Comment by vancluever** [comment]
This seems to have progressed to the point where the non-HTTPS redirect URL (http://releases.rancher.com/os/latest) is now 404ing as well:

```
$ curl -IL http://releases.rancher.com/os/latest
HTTP/1.1 404 Not Found
Date: Fri, 06 Apr 2018 15:29:22 GMT
Content-Type: application/xml; charset=UTF-8
Content-Length: 192
Connection: keep-alive
Set-Cookie: ****
X-GUploader-UploadID: ****
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Content-Type
Expires: Fri, 06 Apr 2018 15:29:22 GMT
Cache-Control: private, max-age=0
Server: cloudflare
CF-RAY: ****
```

**Comment by niusmallnan** [comment]
Hi, @vancluever 
Can you please share your iPXE file?
I have a test setup with Virtualbox, here are my iPXE file:
```
#!ipxe
dhcp
set base-url http://xxxxx/rancheros/ipxe
kernel ${base-url}/vmlinuz rancher.state.autoformat=[/dev/sda] rancher.state.dev=LABEL=RANCHER_STATE rancher.state.wait printk.devkmsg=on console=tty1 rancher.autologin=tty1 rancher.cloud_init.datasources=[url:${base-url}/cloud-config.yaml] rancher.password=rancher
initrd ${base-url}/initrd
boot
```

**Comment by vancluever** [comment]
@niusmallnan that iPXE file is different than the one documented on https://rancher.com/docs/os/v1.2/en/running-rancheros/server/pxe/. If it's been updated, can you all update the docs?

FYI: We have a local build on our TFTP server, so that's not a problem. However, I want to use upstream sources whenever possible.

Thanks!

**Comment by niusmallnan** [comment]
When I set the base-url to `http://releases.rancher.com/os/latest`, I didn't get a 404 error.
In fact it needs `vmlinuz ` and `initrd`.
vmlinuz: http://releases.rancher.com/os/latest/vmlinuz
initrd: http://releases.rancher.com/os/latest/initrd

There is no fixed iPXE file, depending on which kernel parameters you need to set.

**Comment by vancluever** [comment]
@niusmallnan right - I it slipped my mind that the base URL is just a variable and the full URL should have been checked. Sorry!

In any case, I just switched the sources back and the iPXE error is resolved, so I'm going to close this now.

Thanks for your help!

PS: The iPXE error code is `ENOTSUP_ALGORITHM`, which happens as well when I switch `base-url` to the HTTPS source. I noticed that it was redirecting to HTTPS when this error came up and then it stopped, so maybe it has to do with that. I'm using the iPXE bundled with Ubuntu 16.04 and given the version I'm guessing it's using deprecated crypto, which would explain the error.
